### ISSUE TITLE
Exception when changing between scrollable and non-scrollable TabBar

### ISSUE THREAD

**Issue by vonkez** [flutter/flutter#120649]
## Steps to Reproduce

1. Execute `flutter run` on the code sample 
2. Tap on + button 3 times and you willl see the error

**Expected results:** TabBar should change

**Actual results:** TabBar doesn't change instead you get an error 

<details>
<summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(const MyHomePage());
}

class MyHomePage extends StatefulWidget {
  const MyHomePage({super.key});

  @override
  State<MyHomePage> createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> with TickerProviderStateMixin {
  int index = 0;
  late List<TabBar> tabBars;

  @override
  void initState() {
    super.initState();
    tabBars = [
      TabBar(
          controller: TabController(length: 2, vsync: this),
          tabs: const [Tab(text: "AAAAAAAAAAAAAAAAAAAA"), Tab(text: "BBBBBBBBBBBBBBBBBBBBBBBB")]
      ),
      TabBar(
          isScrollable: true,
          controller: TabController(length: 2, vsync: this),
          tabs: const [Tab(text: "CCCCCCCCCCCCCCCCCCC"), Tab(text: "DDDDDDDDDDDDDDDDDDDDD")]
      ),
    ];
  }

  void _changeTabBar() {
    setState(() {
      index = index == 0 ? 1 : 0;
    });
  }

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: Scaffold(
        appBar: AppBar(
          title: const Text("Tab test"),
          bottom: tabBars[index],
        ),
        body: TabBarView(
          controller: tabBars[index].controller,
          children: const [Center(child: Text("PAGE1")), Center(child: Text("PAGE2"))],
        ),
        floatingActionButton: FloatingActionButton(
          onPressed: _changeTabBar,
          child: const Icon(Icons.add),
        ),
      ),
    );
  }
}

```

</details>

<details>
  <summary>Logs</summary>

```
 ══╡ EXCEPTION CAUGHT BY WIDGETS LIBRARY ╞═══════════════════════════════════════════════════════════
                     The following assertion was thrown building MediaQuery(MediaQueryData(size: Size(411.4, 683.4),
                     devicePixelRatio: 2.6, textScaleFactor: 1.0, platformBrightness: Brightness.light, padding:
                     EdgeInsets.zero, viewPadding: EdgeInsets.zero, viewInsets: EdgeInsets.zero, alwaysUse24HourFormat:
                     false, accessibleNavigation: false, highContrast: false, disableAnimations: false, invertColors:
                     false, boldText: false, navigationMode: traditional, gestureSettings:
                     DeviceGestureSettings(touchSlop: 8.0), displayFeatures: [])):
                     ScrollController not attached to any scroll views.
                     'package:flutter/src/widgets/scroll_controller.dart':
                     Failed assertion: line 107 pos 12: '_positions.isNotEmpty'

                     Either the assertion indicates an error in the framework itself, or we should provide substantially
                     more information in this error message to help you determine and fix the underlying cause.
                     In either case, please report this assertion by filing a bug on GitHub:
                       https://github.com/flutter/flutter/issues/new?template=2_bug.md

                     The relevant error-causing widget was:
                       AppBar AppBar:file:///F:/dev/projects/tabtest/lib/main.dart:44:17

                     When the exception was thrown, this was the stack:
                     #2      ScrollController.position (package:flutter/src/widgets/scroll_controller.dart:107:12)
                     #3      _TabBarState.didUpdateWidget (package:flutter/src/material/tabs.dart:1085:60)
                     #4      StatefulElement.update (package:flutter/src/widgets/framework.dart:5142:55)
                     #5      Element.updateChild (package:flutter/src/widgets/framework.dart:3660:15)
                     #6      RenderObjectElement.updateChildren (package:flutter/src/widgets/framework.dart:5981:32)
                     #7      MultiChildRenderObjectElement.update (package:flutter/src/widgets/framework.dart:6537:17)
                     #8      Element.updateChild (package:flutter/src/widgets/framework.dart:3660:15)
                     #9      ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4993:16)
                     #10     Element.rebuild (package:flutter/src/widgets/framework.dart:4690:5)
                     #11     ProxyElement.update (package:flutter/src/widgets/framework.dart:5301:5)
                     #12     Element.updateChild (package:flutter/src/widgets/framework.dart:3660:15)
                     #13     SingleChildRenderObjectElement.update (package:flutter/src/widgets/framework.dart:6384:14)
                     #14     Element.updateChild (package:flutter/src/widgets/framework.dart:3660:15)
                     #15     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4993:16)
                     #16     Element.rebuild (package:flutter/src/widgets/framework.dart:4690:5)
                     #17     StatelessElement.update (package:flutter/src/widgets/framework.dart:5044:5)
                     #18     Element.updateChild (package:flutter/src/widgets/framework.dart:3660:15)
                     #19     SingleChildRenderObjectElement.update (package:flutter/src/widgets/framework.dart:6384:14)
                     #20     Element.updateChild (package:flutter/src/widgets/framework.dart:3660:15)
                     #21     SingleChildRenderObjectElement.update (package:flutter/src/widgets/framework.dart:6384:14)
                     #22     Element.updateChild (package:flutter/src/widgets/framework.dart:3660:15)
                     #23     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4993:16)
                     #24     Element.rebuild (package:flutter/src/widgets/framework.dart:4690:5)
                     #25     ProxyElement.update (package:flutter/src/widgets/framework.dart:5301:5)
                     #26     Element.updateChild (package:flutter/src/widgets/framework.dart:3660:15)
                     #27     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4993:16)
                     #28     StatefulElement.performRebuild (package:flutter/src/widgets/framework.dart:5133:11)
                     #29     Element.rebuild (package:flutter/src/widgets/framework.dart:4690:5)
                     #30     StatefulElement.update (package:flutter/src/widgets/framework.dart:5156:5)
                     #31     Element.updateChild (package:flutter/src/widgets/framework.dart:3660:15)
                     #32     SingleChildRenderObjectElement.update (package:flutter/src/widgets/framework.dart:6384:14)
                     #33     Element.updateChild (package:flutter/src/widgets/framework.dart:3660:15)
                     #34     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4993:16)
                     #35     Element.rebuild (package:flutter/src/widgets/framework.dart:4690:5)
                     #36     ProxyElement.update (package:flutter/src/widgets/framework.dart:5301:5)
                     #37     Element.updateChild (package:flutter/src/widgets/framework.dart:3660:15)
                     #38     SingleChildRenderObjectElement.update (package:flutter/src/widgets/framework.dart:6384:14)
                     #39     Element.updateChild (package:flutter/src/widgets/framework.dart:3660:15)
                     #40     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4993:16)
                     #41     StatefulElement.performRebuild (package:flutter/src/widgets/framework.dart:5133:11)
                     #42     Element.rebuild (package:flutter/src/widgets/framework.dart:4690:5)
                     #43     StatefulElement.update (package:flutter/src/widgets/framework.dart:5156:5)
                     #44     Element.updateChild (package:flutter/src/widgets/framework.dart:3660:15)
                     #45     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4993:16)
                     #46     StatefulElement.performRebuild (package:flutter/src/widgets/framework.dart:5133:11)
                     #47     Element.rebuild (package:flutter/src/widgets/framework.dart:4690:5)
                     #48     StatefulElement.update (package:flutter/src/widgets/framework.dart:5156:5)
                     #49     Element.updateChild (package:flutter/src/widgets/framework.dart:3660:15)
                     #50     SingleChildRenderObjectElement.update (package:flutter/src/widgets/framework.dart:6384:14)
                     #51     Element.updateChild (package:flutter/src/widgets/framework.dart:3660:15)
                     #52     SingleChildRenderObjectElement.update (package:flutter/src/widgets/framework.dart:6384:14)
                     #53     Element.updateChild (package:flutter/src/widgets/framework.dart:3660:15)
                     #54     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4993:16)
                     #55     StatefulElement.performRebuild (package:flutter/src/widgets/framework.dart:5133:11)
                     #56     Element.rebuild (package:flutter/src/widgets/framework.dart:4690:5)
                     #57     StatefulElement.update (package:flutter/src/widgets/framework.dart:5156:5)
                     #58     Element.updateChild (package:flutter/src/widgets/framework.dart:3660:15)
                     #59     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4993:16)
                     #60     Element.rebuild (package:flutter/src/widgets/framework.dart:4690:5)
                     #61     ProxyElement.update (package:flutter/src/widgets/framework.dart:5301:5)
                     #62     Element.updateChild (package:flutter/src/widgets/framework.dart:3660:15)
                     #63     SingleChildRenderObjectElement.update (package:flutter/src/widgets/framework.dart:6384:14)
                     #64     Element.updateChild (package:flutter/src/widgets/framework.dart:3660:15)
                     #65     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4993:16)
                     #66     Element.rebuild (package:flutter/src/widgets/framework.dart:4690:5)
                     #67     ProxyElement.update (package:flutter/src/widgets/framework.dart:5301:5)
                     #68     Element.updateChild (package:flutter/src/widgets/framework.dart:3660:15)
                     #69     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4993:16)
                     #70     Element.rebuild (package:flutter/src/widgets/framework.dart:4690:5)
                     #71     ProxyElement.update (package:flutter/src/widgets/framework.dart:5301:5)
                     #72     Element.updateChild (package:flutter/src/widgets/framework.dart:3660:15)
                     #73     RenderObjectElement.updateChildren (package:flutter/src/widgets/framework.dart:5981:32)
                     #74     MultiChildRenderObjectElement.update (package:flutter/src/widgets/framework.dart:6537:17)
                     #75     Element.updateChild (package:flutter/src/widgets/framework.dart:3660:15)
                     #76     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4993:16)
                     #77     Element.rebuild (package:flutter/src/widgets/framework.dart:4690:5)
                     #78     ProxyElement.update (package:flutter/src/widgets/framework.dart:5301:5)
                     #79     Element.updateChild (package:flutter/src/widgets/framework.dart:3660:15)
                     #80     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4993:16)
                     #81     StatefulElement.performRebuild (package:flutter/src/widgets/framework.dart:5133:11)
                     #82     Element.rebuild (package:flutter/src/widgets/framework.dart:4690:5)
                     #83     StatefulElement.update (package:flutter/src/widgets/framework.dart:5156:5)
                     #84     Element.updateChild (package:flutter/src/widgets/framework.dart:3660:15)
                     #85     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4993:16)
                     #86     StatefulElement.performRebuild (package:flutter/src/widgets/framework.dart:5133:11)
                     #87     Element.rebuild (package:flutter/src/widgets/framework.dart:4690:5)
                     #88     StatefulElement.update (package:flutter/src/widgets/framework.dart:5156:5)
                     #89     Element.updateChild (package:flutter/src/widgets/framework.dart:3660:15)
                     #90     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4993:16)
                     #91     Element.rebuild (package:flutter/src/widgets/framework.dart:4690:5)
                     #92     ProxyElement.update (package:flutter/src/widgets/framework.dart:5301:5)
                     #93     Element.updateChild (package:flutter/src/widgets/framework.dart:3660:15)
                     #94     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4993:16)
                     #95     StatefulElement.performRebuild (package:flutter/src/widgets/framework.dart:5133:11)
                     #96     Element.rebuild (package:flutter/src/widgets/framework.dart:4690:5)
                     #97     StatefulElement.update (package:flutter/src/widgets/framework.dart:5156:5)
                     #98     Element.updateChild (package:flutter/src/widgets/framework.dart:3660:15)
                     #99     SingleChildRenderObjectElement.update (package:flutter/src/widgets/framework.dart:6384:14)
                     #100    Element.updateChild (package:flutter/src/widgets/framework.dart:3660:15)
                     #101    ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4993:16)
                     #102    Element.rebuild (package:flutter/src/widgets/framework.dart:4690:5)
                     #103    ProxyElement.update (package:flutter/src/widgets/framework.dart:5301:5)
                     #104    Element.updateChild (package:flutter/src/widgets/framework.dart:3660:15)
                     #105    SingleChildRenderObjectElement.update (package:flutter/src/widgets/framework.dart:6384:14)
                     #106    Element.updateChild (package:flutter/src/widgets/framework.dart:3660:15)
                     #107    ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4993:16)
                     #108    StatefulElement.performRebuild (package:flutter/src/widgets/framework.dart:5133:11)
                     #109    Element.rebuild (package:flutter/src/widgets/framework.dart:4690:5)
                     #110    StatefulElement.update (package:flutter/src/widgets/framework.dart:5156:5)
                     #111    Element.updateChild (package:flutter/src/widgets/framework.dart:3660:15)
                     #112    ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4993:16)
                     #113    StatefulElement.performRebuild (package:flutter/src/widgets/framework.dart:5133:11)
                     #114    Element.rebuild (package:flutter/src/widgets/framework.dart:4690:5)
                     #115    StatefulElement.update (package:flutter/src/widgets/framework.dart:5156:5)
                     #116    Element.updateChild (package:flutter/src/widgets/framework.dart:3660:15)
                     #117    ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4993:16)
                     #118    Element.rebuild (package:flutter/src/widgets/framework.dart:4690:5)
                     #119    ProxyElement.update (package:flutter/src/widgets/framework.dart:5301:5)
                     #120    Element.updateChild (package:flutter/src/widgets/framework.dart:3660:15)
                     #121    ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4993:16)
                     #122    Element.rebuild (package:flutter/src/widgets/framework.dart:4690:5)
                     #123    ProxyElement.update (package:flutter/src/widgets/framework.dart:5301:5)
                     #124    Element.updateChild (package:flutter/src/widgets/framework.dart:3660:15)
                     #125    ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4993:16)
                     #126    Element.rebuild (package:flutter/src/widgets/framework.dart:4690:5)
                     #127    ProxyElement.update (package:flutter/src/widgets/framework.dart:5301:5)
                     #128    Element.updateChild (package:flutter/src/widgets/framework.dart:3660:15)
                     #129    ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4993:16)
                     #130    StatefulElement.performRebuild (package:flutter/src/widgets/framework.dart:5133:11)
                     #131    Element.rebuild (package:flutter/src/widgets/framework.dart:4690:5)
                     #132    StatefulElement.update (package:flutter/src/widgets/framework.dart:5156:5)
                     #133    Element.updateChild (package:flutter/src/widgets/framework.dart:3660:15)
                     #134    ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4993:16)
                     #135    Element.rebuild (package:flutter/src/widgets/framework.dart:4690:5)
                     #136    ProxyElement.update (package:flutter/src/widgets/framework.dart:5301:5)
                     #137    Element.updateChild (package:flutter/src/widgets/framework.dart:3660:15)
                     #138    ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4993:16)
                     #139    StatefulElement.performRebuild (package:flutter/src/widgets/framework.dart:5133:11)
                     #140    Element.rebuild (package:flutter/src/widgets/framework.dart:4690:5)
                     #141    StatefulElement.update (package:flutter/src/widgets/framework.dart:5156:5)
                     #142    Element.updateChild (package:flutter/src/widgets/framework.dart:3660:15)
                     #143    SingleChildRenderObjectElement.update (package:flutter/src/widgets/framework.dart:6384:14)
                     #144    Element.updateChild (package:flutter/src/widgets/framework.dart:3660:15)
                     #145    ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4993:16)
                     #146    Element.rebuild (package:flutter/src/widgets/framework.dart:4690:5)
                     #147    StatelessElement.update (package:flutter/src/widgets/framework.dart:5044:5)
                     #148    Element.updateChild (package:flutter/src/widgets/framework.dart:3660:15)
                     #149    SingleChildRenderObjectElement.update (package:flutter/src/widgets/framework.dart:6384:14)
                     #150    Element.updateChild (package:flutter/src/widgets/framework.dart:3660:15)
                     #151    SingleChildRenderObjectElement.update (package:flutter/src/widgets/framework.dart:6384:14)
                     #152    Element.updateChild (package:flutter/src/widgets/framework.dart:3660:15)
                     #153    ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4993:16)
                     #154    StatefulElement.performRebuild (package:flutter/src/widgets/framework.dart:5133:11)
                     #155    Element.rebuild (package:flutter/src/widgets/framework.dart:4690:5)
                     #156    StatefulElement.update (package:flutter/src/widgets/framework.dart:5156:5)
                     #157    Element.updateChild (package:flutter/src/widgets/framework.dart:3660:15)
                     #158    SingleChildRenderObjectElement.update (package:flutter/src/widgets/framework.dart:6384:14)
                     #159    Element.updateChild (package:flutter/src/widgets/framework.dart:3660:15)
                     #160    ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4993:16)
                     #161    StatefulElement.performRebuild (package:flutter/src/widgets/framework.dart:5133:11)
                     #162    Element.rebuild (package:flutter/src/widgets/framework.dart:4690:5)
                     #163    StatefulElement.update (package:flutter/src/widgets/framework.dart:5156:5)
                     #164    Element.updateChild (package:flutter/src/widgets/framework.dart:3660:15)
                     #165    SingleChildRenderObjectElement.update (package:flutter/src/widgets/framework.dart:6384:14)
                     #166    Element.updateChild (package:flutter/src/widgets/framework.dart:3660:15)
                     #167    ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4993:16)
                     #168    StatefulElement.performRebuild (package:flutter/src/widgets/framework.dart:5133:11)
                     #169    Element.rebuild (package:flutter/src/widgets/framework.dart:4690:5)
                     #170    StatefulElement.update (package:flutter/src/widgets/framework.dart:5156:5)
                     #171    Element.updateChild (package:flutter/src/widgets/framework.dart:3660:15)
                     #172    ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4993:16)
                     #173    StatefulElement.performRebuild (package:flutter/src/widgets/framework.dart:5133:11)
                     #174    Element.rebuild (package:flutter/src/widgets/framework.dart:4690:5)
                     #175    StatefulElement.update (package:flutter/src/widgets/framework.dart:5156:5)
                     #176    Element.updateChild (package:flutter/src/widgets/framework.dart:3660:15)
                     #177    SingleChildRenderObjectElement.update (package:flutter/src/widgets/framework.dart:6384:14)
                     #178    Element.updateChild (package:flutter/src/widgets/framework.dart:3660:15)
                     #179    ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4993:16)
                     #180    StatefulElement.performRebuild (package:flutter/src/widgets/framework.dart:5133:11)
                     #181    Element.rebuild (package:flutter/src/widgets/framework.dart:4690:5)
                     #182    StatefulElement.update (package:flutter/src/widgets/framework.dart:5156:5)
                     #183    Element.updateChild (package:flutter/src/widgets/framework.dart:3660:15)
                     #184    SingleChildRenderObjectElement.update (package:flutter/src/widgets/framework.dart:6384:14)
                     #185    Element.updateChild (package:flutter/src/widgets/framework.dart:3660:15)
                     #186    ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4993:16)
                     #187    StatefulElement.performRebuild (package:flutter/src/widgets/framework.dart:5133:11)
                     #188    Element.rebuild (package:flutter/src/widgets/framework.dart:4690:5)
                     #189    StatefulElement.update (package:flutter/src/widgets/framework.dart:5156:5)
                     #190    Element.updateChild (package:flutter/src/widgets/framework.dart:3660:15)
                     #191    ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4993:16)
                     #192    StatefulElement.performRebuild (package:flutter/src/widgets/framework.dart:5133:11)
                     #193    Element.rebuild (package:flutter/src/widgets/framework.dart:4690:5)
                     #194    StatefulElement.update (package:flutter/src/widgets/framework.dart:5156:5)
                     #195    Element.updateChild (package:flutter/src/widgets/framework.dart:3660:15)
                     #196    ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4993:16)
                     #197    Element.rebuild (package:flutter/src/widgets/framework.dart:4690:5)
                     #198    StatelessElement.update (package:flutter/src/widgets/framework.dart:5044:5)
                     #199    Element.updateChild (package:flutter/src/widgets/framework.dart:3660:15)
                     #200    ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4993:16)
                     #201    StatefulElement.performRebuild (package:flutter/src/widgets/framework.dart:5133:11)
                     #202    Element.rebuild (package:flutter/src/widgets/framework.dart:4690:5)
                     #203    StatefulElement.update (package:flutter/src/widgets/framework.dart:5156:5)
                     #204    Element.updateChild (package:flutter/src/widgets/framework.dart:3660:15)
                     #205    SingleChildRenderObjectElement.update (package:flutter/src/widgets/framework.dart:6384:14)
                     #206    Element.updateChild (package:flutter/src/widgets/framework.dart:3660:15)
                     #207    ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4993:16)
                     #208    Element.rebuild (package:flutter/src/widgets/framework.dart:4690:5)
                     #209    ProxyElement.update (package:flutter/src/widgets/framework.dart:5301:5)
                     #210    _InheritedNotifierElement.update (package:flutter/src/widgets/inherited_notifier.dart:107:11)
                     #211    Element.updateChild (package:flutter/src/widgets/framework.dart:3660:15)
                     #212    SingleChildRenderObjectElement.update (package:flutter/src/widgets/framework.dart:6384:14)
                     #213    Element.updateChild (package:flutter/src/widgets/framework.dart:3660:15)
                     #214    ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4993:16)
                     #215    StatefulElement.performRebuild (package:flutter/src/widgets/framework.dart:5133:11)
                     #216    Element.rebuild (package:flutter/src/widgets/framework.dart:4690:5)
                     #217    StatefulElement.update (package:flutter/src/widgets/framework.dart:5156:5)
                     #218    Element.updateChild (package:flutter/src/widgets/framework.dart:3660:15)
                     #219    ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4993:16)
                     #220    Element.rebuild (package:flutter/src/widgets/framework.dart:4690:5)
                     #221    ProxyElement.update (package:flutter/src/widgets/framework.dart:5301:5)
                     #222    Element.updateChild (package:flutter/src/widgets/framework.dart:3660:15)
                     #223    ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4993:16)
                     #224    Element.rebuild (package:flutter/src/widgets/framework.dart:4690:5)
                     #225    ProxyElement.update (package:flutter/src/widgets/framework.dart:5301:5)
                     #226    Element.updateChild (package:flutter/src/widgets/framework.dart:3660:15)
                     #227    ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4993:16)
                     #228    StatefulElement.performRebuild (package:flutter/src/widgets/framework.dart:5133:11)
                     #229    Element.rebuild (package:flutter/src/widgets/framework.dart:4690:5)
                     #230    StatefulElement.update (package:flutter/src/widgets/framework.dart:5156:5)
                     #231    Element.updateChild (package:flutter/src/widgets/framework.dart:3660:15)
                     #232    ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4993:16)
                     #233    Element.rebuild (package:flutter/src/widgets/framework.dart:4690:5)
                     #234    StatelessElement.update (package:flutter/src/widgets/framework.dart:5044:5)
                     #235    Element.updateChild (package:flutter/src/widgets/framework.dart:3660:15)
                     #236    ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4993:16)
                     #237    Element.rebuild (package:flutter/src/widgets/framework.dart:4690:5)
                     #238    StatelessElement.update (package:flutter/src/widgets/framework.dart:5044:5)
                     #255    ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4993:16)
                     #256    StatefulElement.performRebuild (package:flutter/src/widgets/framework.dart:5133:11)
                     #257    Element.rebuild (package:flutter/src/widgets/framework.dart:4690:5)
                     #258    StatefulElement.update (package:flutter/src/widgets/framework.dart:5156:5)
                     #259    Element.updateChild (package:flutter/src/widgets/framework.dart:3660:15)
                     #260    ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4993:16)
                     #261    StatefulElement.performRebuild (package:flutter/src/widgets/framework.dart:5133:11)
                     #262    Element.rebuild (package:flutter/src/widgets/framework.dart:4690:5)
                     #263    BuildOwner.buildScope (package:flutter/src/widgets/framework.dart:2743:19)
                     #264    WidgetsBinding.drawFrame (package:flutter/src/widgets/binding.dart:863:21)
                     #265    RendererBinding._handlePersistentFrameCallback (package:flutter/src/rendering/binding.dart:381:5)
                     #266    SchedulerBinding._invokeFrameCallback (package:flutter/src/scheduler/binding.dart:1289:15)
                     #267    SchedulerBinding.handleDrawFrame (package:flutter/src/scheduler/binding.dart:1218:9)
                     #268    SchedulerBinding._handleDrawFrame (package:flutter/src/scheduler/binding.dart:1076:5)
                     #269    _invoke (dart:ui/hooks.dart:145:13)
                     #270    PlatformDispatcher._drawFrame (dart:ui/platform_dispatcher.dart:338:5)
                     #271    _drawFrame (dart:ui/hooks.dart:112:31)
                     (elided 2 frames from class _AssertionError)

                     ════════════════════════════════════════════════════════════════════════════════════════════════════

```

```
Analyzing tabtest...                                                    
No issues found! (ran in 1.0s)
```

```
[√] Flutter (Channel stable, 3.7.3, on Microsoft Windows [Version 10.0.19044.2486], locale en-GB)
    • Flutter version 3.7.3 on channel stable at C:\Users\lil\Documents\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 9944297138 (5 days ago), 2023-02-08 15:46:04 -0800
    • Engine revision 248290d6d5
    • Dart version 2.19.2
    • DevTools version 2.20.1

[√] Windows Version (Installed version of Windows is version 10 or higher)

[√] Android toolchain - develop for Android devices (Android SDK version 32.0.0)
    • Android SDK at C:\Users\lil\AppData\Local\Android\sdk
    • Platform android-33, build-tools 32.0.0
    • Java binary at: F:\dev\android studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 11.0.15+0-b2043.56-9505619)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop for Windows (Visual Studio Community 2022 17.0.5)
    • Visual Studio at C:\Program Files\Microsoft Visual Studio\2022\Community
    • Visual Studio Community 2022 version 17.0.32112.339
    • Windows 10 SDK version 10.0.19041.0

[√] Android Studio (version 2022.1)
    • Android Studio at F:\dev\android studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.15+0-b2043.56-9505619)

[√] VS Code (version 1.74.2)
    • VS Code at C:\Users\lil\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension can be installed from:
       https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter

[√] Connected device (4 available)
    • sdk gphone x86 (mobile) • emulator-5554 • android-x86    • Android 11 (API 30) (emulator)
    • Windows (desktop)       • windows       • windows-x64    • Microsoft Windows [Version 10.0.19044.2486]
    • Chrome (web)            • chrome        • web-javascript • Google Chrome 109.0.5414.120
    • Edge (web)              • edge          • web-javascript • Microsoft Edge 110.0.1587.41

[√] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!

```

</details>

### Video

[tabbar_bug.webm](https://user-images.githubusercontent.com/15914944/218610659-a877cb7f-4c24-463e-bbd5-c8ab6ceff7ad.webm)

**Comment by exaby73** [comment]
## Triage report

I can reproduce this issue with `Master (3.8.0-12.0.pre.50)`. Using a single `TabBar` with dynamically changing the `isScrollable` property does not reproduce this issue. Looks like this only reproduces with multiple `TabBar`s.

<details><summary>Code Sample (Same as OP)</summary>

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(const MyHomePage());
}

class MyHomePage extends StatefulWidget {
  const MyHomePage({super.key});

  @override
  State<MyHomePage> createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> with TickerProviderStateMixin {
  int index = 0;
  late List<TabBar> tabBars;

  @override
  void initState() {
    super.initState();
    tabBars = [
      TabBar(
          controller: TabController(length: 2, vsync: this),
          tabs: const [Tab(text: "AAAAAAAAAAAAAAAAAAAA"), Tab(text: "BBBBBBBBBBBBBBBBBBBBBBBB")]
      ),
      TabBar(
          isScrollable: true,
          controller: TabController(length: 2, vsync: this),
          tabs: const [Tab(text: "CCCCCCCCCCCCCCCCCCC"), Tab(text: "DDDDDDDDDDDDDDDDDDDDD")]
      ),
    ];
  }

  void _changeTabBar() {
    setState(() {
      index = index == 0 ? 1 : 0;
    });
  }

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: Scaffold(
        appBar: AppBar(
          title: const Text("Tab test"),
          bottom: tabBars[index],
        ),
        body: TabBarView(
          controller: tabBars[index].controller,
          children: const [Center(child: Text("PAGE1")), Center(child: Text("PAGE2"))],
        ),
        floatingActionButton: FloatingActionButton(
          onPressed: _changeTabBar,
          child: const Icon(Icons.add),
        ),
      ),
    );
  }
}
```

</details>

<details><summary><code>flutter doctor -v</code> (Master)</summary>

```
[!] Flutter (Channel master, 3.8.0-12.0.pre.50, on macOS 13.2 22D49 darwin-arm64, locale en-US)
    • Flutter version 3.8.0-12.0.pre.50 on channel master at /Users/nabeelparkar/fvm/versions/master
    ! Warning: `dart` on your path resolves to /opt/homebrew/Cellar/dart/2.19.1/libexec/bin/dart, which is not inside your current Flutter SDK checkout at /Users/nabeelparkar/fvm/versions/master. Consider adding /Users/nabeelparkar/fvm/versions/master/bin to the front of your path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 6a94f25a9d (14 minutes ago), 2023-02-14 02:22:14 -0500
    • Engine revision 17ab09d382
    • Dart version 3.0.0 (build 3.0.0-233.0.dev)
    • DevTools version 2.21.1
    • If those were intentional, you can disregard the above warnings; however it is recommended to use "git" directly to perform update checks and upgrades.

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.0)
    • Android SDK at /Users/nabeelparkar/Library/Android/sdk/
    • Platform android-33, build-tools 33.0.0
    • ANDROID_SDK_ROOT = /Users/nabeelparkar/Library/Android/sdk/
    • Java binary at: /Users/nabeelparkar/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/221.6008.13.2211.9477386/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.15+0-b2043.56-8887301)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.2)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14C18
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • CHROME_EXECUTABLE = /Applications/Brave Browser.app/Contents/MacOS/Brave Browser

[✓] Android Studio (version 2021.3)
    • Android Studio at /Users/nabeelparkar/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/213.7172.25.2113.9123335/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)

[✓] Android Studio (version 2022.1)
    • Android Studio at /Users/nabeelparkar/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/221.6008.13.2211.9477386/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.15+0-b2043.56-8887301)

[✓] VS Code (version 1.75.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.58.0

[✓] Connected device (2 available)
    • macOS (desktop) • macos  • darwin-arm64   • macOS 13.2 22D49 darwin-arm64
    • Chrome (web)    • chrome • web-javascript • Brave Browser 110.1.48.158

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 1 category.
```

</details>

**Comment by vonkez** [comment]
I also noticed this issue doesn't exists on version 3.3.10. It's started after version 3.7.
